torchmetrics/functional/text/rouge.py

```diff
@@ -12,114 +12,229 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import re
-from typing import Dict, List, Tuple, Union
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Tuple, Union
 
 import torch
-from torch import Tensor, tensor
-
-from torchmetrics.utilities.imports import _NLTK_AVAILABLE, _ROUGE_SCORE_AVAILABLE
-
-if _ROUGE_SCORE_AVAILABLE:
-    from rouge_score.rouge_scorer import RougeScorer
-    from rouge_score.scoring import AggregateScore, BootstrapAggregator
-else:
-    RougeScorer, AggregateScore, BootstrapAggregator = object, object, object
-
-ALLOWED_ROUGE_KEYS = (
-    "rouge1",
-    "rouge2",
-    "rouge3",
-    "rouge4",
-    "rouge5",
-    "rouge6",
-    "rouge7",
-    "rouge8",
-    "rouge9",
-    "rougeL",
-    "rougeLsum",
-)
-
-
-def add_newline_to_end_of_each_sentence(x: str) -> str:
+from torch import Tensor
+
+from torchmetrics.utilities.imports import _NLTK_AVAILABLE
+
+if _NLTK_AVAILABLE:
+    import nltk
+
+ALLOWED_ROUGE_KEYS: Dict[str, Union[int, str]] = {
+    "rouge1": 1,
+    "rouge2": 2,
+    "rouge3": 3,
+    "rouge4": 4,
+    "rouge5": 5,
+    "rouge6": 6,
+    "rouge7": 7,
+    "rouge8": 8,
+    "rouge9": 9,
+    "rougeL": "L",
+    "rougeLsum": "Lsum",
+}
+
+
+@dataclass
+class _RougeScore:
```

Comment:
Why do you need a specific class for it? Doesn't a dict work here?

Reply:
Yes, `dict` should be sufficient here. I will replace that!